src/carma-streets/parse_kafka_logs.py

```diff
@@ -45,20 +48,24 @@ def parse_kafka_logs_as_type(input_file_path: Path, message_type: KafkaLogMessag
         with open(input_file_path, encoding='utf8', errors='ignore') as log_file:
             msgs = []
             skipped_messages = 0
-            for line in log_file:
+            kafka_message = str()
+            for line_number, line in enumerate(log_file):
                 try:
                     line = line.strip()
                     #get the create time stamped by kafka
                     create_index = line.find('CreateTime')
                     if (create_index != -1):
+                        if kafka_message:
+                            kafka_json = json.loads(kafka_message)
+                            kafka_message = ''
+                            msgs.append(KafkaLogMessage(create_time, kafka_json, message_type))
                         create_time = re.sub('[^0-9]', '', line.split(':')[1])      
-
-                    json_beg_index = line.find('{')
-                    kafka_message = line[json_beg_index:]
-                    kafka_json = json.loads(kafka_message)
-                    msgs.append(KafkaLogMessage(create_time, kafka_json, message_type))
+                        json_beg_index = line.find('{')
+                        kafka_message += line[json_beg_index:]
```

Comment:
Consider abstracting this behind a function to make the code more expressive:

```python
def get_partial_json_payload(line: str):
    pass

kafak_message += get_partial_json_payload(line)
```

Reply:
I think this would be difficult to abstract into a separate function. Depending on if the line is one with the "CreatedTime" field or not the extraction of json changes from `kafka_message += line[bracket index:]` to `kafka_message += line`. I think putting this in a separate function just makes it harder to trace the logic

Reply:
Yeah that's fine. This code section is small enough.